catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java

```diff
@@ -324,38 +308,45 @@ public UpdateResponse update(UpdateRequest updateRequest) throws IngestException
 
     /* 1. QUERY */
     // Loop to get all identifiers
-    List<String> identifiers =
-        updates
-            .stream()
-            .map(Entry::getKey)
-            .map(Serializable::toString)
-            .collect(Collectors.toList());
+    Set<String> identifiers =
+        updates.stream().map(Entry::getKey).map(Serializable::toString).collect(Collectors.toSet());
 
-    /* 1a. Create the old Metacard Query */
-    String attributeQuery = getQuery(attributeName, identifiers);
-
-    SolrQuery query = new SolrQuery(attributeQuery);
-    // Set number of rows to the result size + 1.  The default row size in Solr is 10, so this
-    // needs to be set in situations where the number of metacards to update is > 10.  Since there
-    // could be more results in the query response than the number of metacards in the update
-    // request,
-    // 1 is added to the row size, so we can still determine whether we found more metacards than
-    // updated metacards provided
-    query.setRows(updates.size() + 1);
-    QueryResponse idResults = null;
+    Map<Serializable, Metacard> idToMetacardMap = new HashMap<>();
+    if (Metacard.ID.equals(attributeName)) {
+      try {
+        idToMetacardMap =
+            client
+                .getIds(identifiers)
+                .stream()
+                .filter(Objects::nonNull)
```

Comment:
❓ What is the case where null ids can occur?

Reply:
Currently it is not possible.  This is more defensive since the contract with SolrMetacardClient currently does not guarantee that the list will not contain nulls.